p2p/security/noise/transport.go

```diff
@@ -65,3 +82,50 @@ func (t *Transport) WithSessionOptions(opts ...SessionOption) (sec.SecureTranspo
 	}
 	return st, nil
 }
+
+func MatchMuxers(initiatorMuxers, responderMuxers []string) string {
+	respMuxerMap := make(map[string]struct{}, len(responderMuxers))
```

Comment:
This has the best scaling behavior for n -> inf, but our n is limited by the protobuf size to a few hundred values (and in practice, will never be more than a single-digit number).

I think we can live with the O(n^2) scaling of just iterating over both lists. This makes this code easier to read, and it avoids the allocation associated with creating a map.